src/app.py

```diff
@@ -95,90 +103,94 @@ def query() -> Any:
         "status": "new",
         "rule_name": rules[-1].name,
         "rule_author": rules[-1].author,
-        "raw_yara": raw_yara,
+        "raw_yara": data.raw_yara,
         "submitted": int(time.time()),
-        "priority": req["priority"],
+        "priority": data.priority,
     }
 
-    if "taint" in req and req["taint"] is not None:
-        job_obj["taint"] = req["taint"]
+    if data.taint is not None:
+        job_obj["taint"] = data.taint
 
     redis.hmset("job:" + job_hash, job_obj)
     redis.rpush("queue-search", job_hash)
 
-    return jsonify({"query_hash": job_hash})
-
+    return QueryResponseSchema(query_hash=job_hash)
 
-@app.route("/api/matches/<hash>")
-def matches(hash: str) -> Response:
-    offset = int(request.args["offset"])
-    limit = int(request.args["limit"])
 
+@app.get("/api/matches/{hash}", response_model=MatchesSchema)
+def matches(
+    hash: str, offset: int = Query(...), limit: int = Query(...)
+) -> MatchesSchema:
     p = redis.pipeline(transaction=False)
     p.hgetall("job:" + hash)
     p.lrange("meta:" + hash, offset, offset + limit - 1)
     job, meta = p.execute()
-
-    return jsonify({"job": job, "matches": [json.loads(m) for m in meta]})
+    return MatchesSchema(job=job, matches=[json.loads(m) for m in meta])
+
+
+def get_job(job_id: str) -> JobSchema:
+    job = redis.hgetall(job_id)
+    return JobSchema(
+        id=job_id[4:],
+        status=job.get("status", "[error]"),
```

Comment:
1) It's a bit counter intuitive to get `"[error"]` in your JSON (string that looks like an array, but contains invalid data). Proposal: `"ERROR"`
2) I don't think returning errors with normal data structure and 200 OK is the way to go (just as a general practice).

Reply:
1. Right

2. I agree, but redis is schemaless, and I've had situations (not yesterday, just in general) where just one invalid record (for example, missing status) caused 500 erorrs on the whole website. Right now we're using redis so we should just embrace the fact that dhe data may or may not be there. As a longer term goal, I hope we migrate to postgres or some other relational database :thinking:.

Reply:
:+1: